src/Core/Utilities/CoreHelpers.cs

```diff
@@ -191,9 +191,9 @@ public static Uri GetUri(string uriString)
                 var queryStringNameValueCollection = HttpUtility.ParseQueryString(queryStringParams);
                 foreach(var key in queryStringNameValueCollection.AllKeys)
                 {
-                    if (!dict.ContainsKey(key))
+                    if (key!= null && !dict.ContainsKey(key))
                     {
-                        dict.Add(key, queryStringNameValueCollection[key] == null ? string.Empty : System.Net.WebUtility.UrlDecode(queryStringNameValueCollection[key]));
+                        dict.Add(key, queryStringNameValueCollection[key]);
                     }
                 }
```

Comment:
As it states in the docs, `ParseQueryString` already takes care of repeating keys that are consolidated into one entry on the `NameValueCollection` with comma separated values, so the if check is not necessary and there can't be `null` keys. So this can be written as:
```c#
return queryStringNameValueCollection.AllKeys.ToDictionary(k => k, k => queryStringNameValueCollection[k]);
```
Then remove creating the dict at the top and return an empty Dictionary in the catch just in case